File: public/examples/integrations/toolkits/box_api/retrieve_file_details_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "BoxApi.RetrieveFileDetails"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=TOOL_NAME
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'file_identifier': '12345',
    'included_file_attributes': ['name', 'size', 'metadata.enterpriseProperties'],
    'etag_conditional_retrieval': '"0a1b2c3d4e5f"',
    'shared_link_with_optional_password': 'shared_link=https://example.app.box.com/s/abcd1234&shared_link_password=pa$$w0rd',
    'file_representations_request': '[jpg?dimensions=128x128][pdf?locale=en_US]'
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/box_api/retrieve_file_details_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "BoxApi.RetrieveFileDetails";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "file_identifier": "12345",
  "included_file_attributes": [
    "name",
    "size",
    "metadata:global.key1"
  ],
  "etag_conditional_retrieval": "0xabcdef1234567890",
  "shared_link_with_optional_password": "shared_link=https://example.app.box.com/s/abc123&shared_link_password=secret",
  "file_representations_request": "[jpg?dimensions=128x128][pdf?max_width=1024]"
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
